aptos-move/e2e-move-tests/src/tests/storage_refund.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, tests::common::test_dir_path, MoveHarness};
+use aptos_language_e2e_tests::account::Account;
+use aptos_types::on_chain_config::FeatureFlag;
+use aptos_vm::testing::{testing_only::inject_error_once, InjectedError};
+use move_core_types::account_address::AccountAddress;
+use serde::Serialize;
+
+#[test]
+fn test_refunds() {
```

Comment:
Can you test this on dropping smart tables? cc @lightmark

Reply:
What specifically are we worried about here? It's no different than a normal resource, right?

Reply:
It's potentially a lot more if the table is very large (e.g. 1000 buckets)?

Reply:
And the table owner gets the refund? Isn't this kinda broken? e.g. I create a smart table where other ppl write stuff into and pay gas, and I drop the table and get a huge refund.

Reply:
@bowenyang007 Yes, that was the debate (and you were one of the believers of refunding to the deleter). People think it makes more sense than the other way around, partly because as a user you trust the contract developer in the first place.

Reply:
Well tables are weird b/c it's not just one person paying gas. It's potentially A LOT of people. How do other chains do it?

Reply:
Just mentioned you in the comparison doc.

Reply:
There are no tables on other chains. Closest to mapping on EVM but you have to delete one slot at a time. It's good to encourage destroying an entire table though such as for airdrop whitelisting as it's just dead data afterward. If the developers leave a way to delete it, they can get a lot more refunds.